feat(update.sh): [Don’t update kernel if kernel version is already installed]

Kreblskulm opened the pull request:
This will make it so that when the update is called it will check for the existence of 5.17.5 in the config files as an identifier for the kernel and if this identifier is found the kernel will not be installed due to it being installed already

Kreblskulm committed:
feat(update.sh)

This change should make it so that you will not need to install the kernal each update

ajstrongdev commented:
Will review when I'm free. Until then I've assigned @wizard-28 to review.

ajstrongdev commented on update.sh:
```diff
@@ -32,12 +32,15 @@ rm -rf ~/rhino-config
+    wget -q --show-progress --progress=bar:force https://kernel.ubuntu.com/~kernel-ppa/mainline/v5.17.5/amd64/linux-image-unsigned-5.17.5-051705-generic_5.17.5-051705.202204271406_amd64.deb
+    wget -q --show-progress --progress=bar:force https://kernel.ubuntu.com/~kernel-ppa/mainline/v5.17.5/amd64/linux-modules-5.17.5-051705-generic_5.17.5-051705.202204271406_amd64.deb
+    sudo apt install ./*.deb
+    : > "$HOME/.rhino/config/5.17.5"
```
The file should've been called 5-17-5. Minimal change but recommended.

ajstrongdev commented on update.sh:
```diff
@@ -32,12 +32,15 @@ rm -rf ~/rhino-config
-  wget -q --show-progress --progress=bar:force https://kernel.ubuntu.com/~kernel-ppa/mainline/v5.17.5/amd64/linux-image-unsigned-5.17.5-051705-generic_5.17.5-051705.202204271406_amd64.deb
-  wget -q --show-progress --progress=bar:force https://kernel.ubuntu.com/~kernel-ppa/mainline/v5.17.5/amd64/linux-modules-5.17.5-051705-generic_5.17.5-051705.202204271406_amd64.deb
-  sudo apt install ./*.deb
+  if [[ ! -f "$HOME/.rhino/config/5.17.5" ]]; then
```
File should be "5-17-5"

ajstrongdev reviewed (changes requested):
Please change the file name to 5-17-5 rather than 5.17.5

Kreblskulm committed:
feat(update.sh): Don’t update kernel if kernel version is already installed

This change should make it so that you will not need to install the kernel each update

ajstrongdev reviewed (approved):
All of the changes requested seem to have been fixed. Will test on Tuesday and then if it works as intended. It'll be merged.

wizard-28 commented on update.sh:
```diff
@@ -61,5 +64,4 @@ fi
-The upgrade has been completed. Please reboot your system to see the changes.
----"
\ No newline at end of file
+The upgrade has been completed. Please reboot your system to see the changes.---"
```
Why is this changed?

ajstrongdev replied:
I am unsure. I'll review it in further detail when I am home.

wizard-28 commented on update.sh:
```diff
@@ -32,12 +32,15 @@ rm -rf ~/rhino-config
-  wget -q --show-progress --progress=bar:force https://kernel.ubuntu.com/~kernel-ppa/mainline/v5.17.5/amd64/linux-image-unsigned-5.17.5-051705-generic_5.17.5-051705.202204271406_amd64.deb
-  wget -q --show-progress --progress=bar:force https://kernel.ubuntu.com/~kernel-ppa/mainline/v5.17.5/amd64/linux-modules-5.17.5-051705-generic_5.17.5-051705.202204271406_amd64.deb
-  sudo apt install ./*.deb
+  if [[ ! -f "$HOME/.rhino/config/5-17-5" ]]; then
```
Wouldn't it be better to check if the packages are already installed on the system by querying APT, instead of creating new files?

ajstrongdev replied:
I just believed this would be the most simple solution. Furthermore as Ubuntu devel upgrades their repositories there may be times where the Ubuntu kernel is installed over the mainline one, with matching version numbers. Wouldn't this create more issues in the long run?

ajstrongdev commented on update.sh:
```diff
@@ -61,5 +64,4 @@ fi
-The upgrade has been completed. Please reboot your system to see the changes.
----"
\ No newline at end of file
+The upgrade has been completed. Please reboot your system to see the changes.---"
```
This needs to be fixed. It should be:
```sh
echo "---
The upgrade has been completed. Please reboot your system to see the changes
---"
```

ajstrongdev reviewed (changes requested):
Minor change required on the last line which tells the user to reboot.

Kreblskulm committed:
fix(update.sh): Fixed message display bug

Fixed message display bug that tells the user to reboot the system

ajstrongdev reviewed (approved):
Seems to have fixed the mistake. Will merge after some testing.
